src/content/docs/en/recipes/analyze-bundle-size.mdx

```diff
@@ -0,0 +1,81 @@
+---
+title: Analyze Bundle Size
+description: Learn how to analyze the bundle generated by Astro.
+i18nReady: true
+type: recipe
+---
+
+import { Steps } from '@astrojs/starlight/components';
+import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro';
+
+Understanding what is a part of an Astro bundle is important for improving site performance. Visualization is often helpful to quickly make sense of what
+the bandle contains, and can give clues as to where changes can be made to reduce the bundle size.
+
+## Recipe
+
+To visualize the bundle, we'll be using the `rollup-plugin-visualizer` library. Documentation on this specific plugin can be found [here](https://github.com/btd/rollup-plugin-visualizer).
+<Steps>
+1. Install `rollup-plugin-visualizer`:
+
+    <PackageManagerTabs>
+        <Fragment slot="npm">
+        ```shell
+        npm install rollup-plugin-visualizer --save-dev
+        ```
+        </Fragment>
+        <Fragment slot="pnpm">
+        ```shell
+        pnpm add rollup-plugin-visualizer --save-dev
+        ```
+        </Fragment>
+        <Fragment slot="yarn">
+        ```shell
+        yarn add rollup-plugin-visualizer --save-dev
+        ```
+        </Fragment>
+    </PackageManagerTabs>
+
+2. Add the plugin to the `astro.config.mjs` file:
+
+    ```js
+    // @ts-check
+    import { defineConfig } from 'astro/config';
+    import { visualizer } from "rollup-plugin-visualizer";
+
+    export default defineConfig({
+    vite: {
+        plugins: [visualizer({
+            emitFile: true,
+            filename: "stats.html",
+        })]
+    }
+    });
+    ```
+3. Finally, run the build command:
+
+    <PackageManagerTabs>
+        <Fragment slot="npm">
+        ```shell
+        npm run build
+        ```
+        </Fragment>
+        <Fragment slot="pnpm">
+        ```shell
+        pnpm build
+        ```
+        </Fragment>
+        <Fragment slot="yarn">
+        ```shell
+        yarn build
+        ```
+        </Fragment>
+    </PackageManagerTabs>
+        :::note
+        If your `output` mode is set to either `hybrid` or `server`, you will have two `stats.html` files. One will be for the client, and the other for the server. Each `stats.html` file will be at the root of the `dist/client` and `dist/server/` directories.
+</Steps>
+
+After building your site, the `stats.html` file will live at the root of your `dist/` directory. From there, you'll be able to see what is a part of the client bundle. 
+
+:::note
+Given Astro's unique approach to hydration, the build isn't necessarily representative of the bundle 
+that the client will receive. The Rollup visualizer shows all dependencies that are used across the site, but it does not break down the bundle size on a per-page basis.
```

Comment:
```suggestion
4. Find the `stats.html` file(s) for your project.

		This will be at the root of your `dist/` directory for entirely static sites and will allow you to see what is included in the bundle.
		
		If your Astro project uses on-demand rendering, you will have two `stats.html` files. One will be for the client, and the other for the server, and each will be located at the root of the `dist/client` and `dist/server/` directories.
		
</Steps>

:::note
Given Astro's unique approach to hydration, the build isn't necessarily representative of the bundle 
that the client will receive. 

The Rollup visualizer shows all dependencies that are used across the site, but it does not break down the bundle size on a per-page basis.
:::
```
Final comment here, because it seems like the last part might be more helpful if included as the final step.